### PR TITLE
Issue 1362 int types

### PR DESCRIPTION
**Pull request by richardjgowers**
Fixes (partially) #1362
Supercedes #1391 

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] AUTHORS
 - [x] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by richardjgowers** [comment]
@rathann if you want you can add your name to AUTHORS to get some credit for your work

**Comment by richardjgowers** [comment]
Oh, and don't squash this so we keep both authors

File: testsuite/MDAnalysisTests/core/test_index_dtype.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""32 bit compat tests
+
+Tests for making sure that integer arrays used for indexing
+use `np.intp`
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/core/test_index_dtype.py]: maybe add here that `np.intp` is a platform independent type  for indexing in numpy.

**Comment by orbeckst** [comment]
@kain88-de (https://github.com/MDAnalysis/mdanalysis/pull/1393#issuecomment-307718573) maybe you or @richardjgowers (or I)  can add @rathann to the authors? He cannot push to this branch and would need a PR against this PR... 

@rathann can you please provide the information for [packages/AUTHORS](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/AUTHORS) (namely, your full name if you want to be listed with it, otherwise any handle that you'd like to use) *as a comment on this PR* so that one of us can quickly add you?

**Comment by orbeckst** [comment]
> @rathann can you please provide the information for packages/AUTHORS (namely, your full name if you want to be listed with it, otherwise any handle that you'd like to use) as a comment on this PR so that one of us can quickly add you?

I found your name in the submitter email and I am adding you to AUTHORS. Welcome as a MDAnalysis contributor (you'll also get a GitHub invite).

**Comment by orbeckst** [comment]
I rebased against develop and added @rathann to AUTHORS.

**Comment by rathann** [comment]
Thank you!

**Pull request merged**